### PR TITLE
Sanitize href props with xss vulnerability V2

### PR DESCRIPTION
**Pull request by AnnMarieW**
Sanitize html props that are vulnerable to xss vulnerability if user data is inserted.

This is the new version based on comments in #999 

Let's keep this as a draft until https://github.com/plotly/dash/issues/2743 is resolved.

### CODE REVIEW

File: src/components/nav/NavLink.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
-import React, {useEffect, useState} from 'react';
+import React, {useEffect, useState, useMemo} from 'react';
```
Comment (tcbegley) [inline src/components/nav/NavLink.js:1]: `useMemo` not used in this file afaict

**Review by tcbegley — COMMENTED** [review]
Looking good so far.

1000th issue / PR! Cool milestone 😄

File: src/components/nav/NavLink.js (hunk 1 above)
Comment (AnnMarieW) [reply src/components/nav/NavLink.js:1]: Right - I didn't do `npm run format` yet.  Still lots of files to change :-)

**Review by tcbegley — APPROVED** [review]
Great stuff, thanks @AnnMarieW!

I made a few small edits:
* Used `pytest.mark.parameterize` inside the integration tests, I think it makes it a little easier to see what is going on in the test + that the same check is being performed for multiple components.
* Some small changes to prop destructuring, e.g. where we pull out `setProps` we no longer need to call `omit` and so on.

**Pull request merged**